ingestion_server/test/unit_tests/test_slack.py

```diff
@@ -47,16 +48,18 @@ def test_message(
 ):
     monkeypatch.setenv("ENVIRONMENT", environment)
     monkeypatch.setenv(slack.SLACK_WEBHOOK, webhook)
-    with mock.patch("requests.post") as mock_post:
+    pook.on()
+    if webhook:
+        mock_post = pook.post(webhook)
         slack._message(text, summary)
-        assert mock_post.called == should_alert
-        if not should_alert:
-            return
-        data = mock_post.call_args.kwargs["json"]
-        assert data["blocks"][0]["text"]["text"] == text
-        assert data["text"] == expected_summary
-        if environment:
-            assert data["username"].endswith(environment.upper())
+        if should_alert:
+            assert mock_post.calls > 0
+            data = json.loads(mock_post.matches[0].body)
+            assert data["blocks"][0]["text"]["text"] == text
+            assert data["text"] == expected_summary
+            if environment:
+                assert data["username"].endswith(environment.upper())
+    pook.off()
```

Comment:
Similar to the other comment, we need `pook` to turn off safely if the test fails. Either use a context manager, the decorator, or the fixtures I shared.